lib/rule-helpers.js

```diff
@@ -0,0 +1,56 @@
+'use babel';
+
+import { get } from 'request-promise';
+
+const DOCUMENTATION_LIFETIME = 86400 * 1000; // 1 day TODO: Configurable?
+const docsRuleCache = new Map();
+let docsLastRetrieved;
+
+export function takeWhile(source, predicate) {
+  const result = [];
+  const { length } = source;
+  let i = 0;
+
+  while (i < length && predicate(source[i], i)) {
+    result.push(source[i]);
+    i += 1;
+  }
+
+  return result;
+}
+
+// Retrieves style guide documentation with cached responses
+export async function getRuleMarkDown(rule) {
+  if (new Date().getTime() - docsLastRetrieved < DOCUMENTATION_LIFETIME) {
```

Comment:
This cache expiration should probably be per-rule, not a global value for the entire cache.